extensions-core/core/runtime/src/main/java/org/apache/camel/quarkus/core/RuntimeBeanRepository.java

```diff
@@ -163,8 +165,13 @@ public <T> T findSingleByType(Class<T> type) {
             } else {
                 handles = container.listAll(type);
             }
+
             if (handles.size() > 0) {
-                return handles.get(0).get();
+                if (handles.stream().filter(h -> !h.getBean().getKind().equals(InjectableBean.Kind.SYNTHETIC)).count() > 0) {
+                    return handles.get(0).get();
+                } else {
+                    return container.select(type, Default.Literal.INSTANCE).get();
```

Comment:
Is it safe to call `get()` if no default bean exists? IIRC it'll throw an exception if not.

Reply:
Remove it.